providers/dnsimple/dnsimpleProvider.go

```diff
@@ -0,0 +1,314 @@
+package dnsimple
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/models"
+	"github.com/StackExchange/dnscontrol/providers"
+	"github.com/StackExchange/dnscontrol/providers/diff"
+	"github.com/miekg/dns/dnsutil"
+
+	dnsimpleapi "github.com/dnsimple/dnsimple-go/dnsimple"
+)
+
+const stateRegistered = "registered"
+
+var defaultNameServerNames = []string{
+	"ns1.dnsimple.com",
+	"ns2.dnsimple.com",
+	"ns3.dnsimple.com",
+	"ns4.dnsimple.com",
+}
+
+type DnsimpleApi struct {
+	AccountToken string // The account access token
+	BaseURL      string // An alternate base URI
+	accountId    string // Account id cache
+}
+
+func (c *DnsimpleApi) GetNameservers(domainName string) ([]*models.Nameserver, error) {
+	nameServerNames, err := c.getNameservers(domainName)
+	if err != nil {
+		return nil, err
+	}
+
+	return models.StringsToNameservers(nameServerNames), nil
+}
+
+func (c *DnsimpleApi) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	corrections := []*models.Correction{}
+
+	records, err := c.getRecords(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	var actual []*models.RecordConfig
+	for _, r := range records {
+		fqdn := r.Name + "." + dc.Name
+		fqdn = strings.TrimLeft(fqdn, ".")
```

Comment:
This trims the dot from the case where `r.Name` is blank (i.e. it's a record on the apex).

I took this approach rather than branching with an if/else on `r.Name == ""`